wire-compiler/src/main/java/com/squareup/wire/java/ServiceFactory.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (C) 2015 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.wire.java;
+
+import com.squareup.javapoet.TypeSpec;
+import com.squareup.wire.model.WireService;
+import java.util.List;
+
+/**
+ * Generates a Java service class for a .proto RPC service. Implementing classes must have a no
+ * arguments constructor.
+ */
+public interface ServiceFactory {
```

Comment:
Just so I'm clear, you're just trying to get the existing tests to pass, right?

I want the whole service generator thing to be ripped out. It's far too presumptuous and limited to be of value. I don't think any apps use it, plus it's about to be obsoleted by Retrofit 2. The existing implementations here should absolutely not be shipping in the compiler when it's done.

> plugins could generate whatever they wanted

I think this is the _only_ path forward. The specialization of services is limiting and we have tons of other code we want to generate that has full access to the object graph.